editor/progress_dialog.cpp

```diff
@@ -230,12 +249,35 @@ void ProgressDialog::end_task(const String &p_task) {
 	}
 }
 
+void ProgressDialog::_time_info_timer_timeout() {
+	for (Map<String, Task>::Element *E = tasks.front(); E; E = E->next()) {
+		Task task = E->get();
+
+		const uint32_t time_spent = OS::get_singleton()->get_ticks_msec() - task.time_started;
+		// Don't display an ETA if less than 4 seconds have passed.
+		// This is because ETA requires some data to be accurate, and 1-3 seconds of data
+		// is just not accurate enough to give an useful ETA.
+		if (time_spent >= 4000 && !Math::is_zero_approx(task.progress->get_value())) {
+			// Avoid division by zero.
+			const uint32_t eta = time_spent / MAX(0.001, task.progress->get_value() / MAX(0.001, task.progress->get_max())) - time_spent;
+			task.time_info->set_text(vformat("%s (ETA: %s)", get_time_string(time_spent * 0.001), get_time_string(eta * 0.001)));
```

Comment:
This should probably be localizable, as ETA is not a universal term. I'm also not sure that it's the correct term in English (it's a time of *arrival*, while here we track completion), though I don't have a better idea. Also no idea how I'd translate this efficiently in French (though we're not too limited in string length so I guess it can be longer).

Reply:
BTW, at some point we'll need to do better than `%s` substitutions if we want to do localization right, since as it stands it wouldn't allow reordering the terms (which might be needed here in a RTL language for example).

Reply:
In most cases %s is ok for RTL. Currently the only string which might have issues with the ordering looks like "something %s, %s and %s", with with LTR names substituted for all %s, but it's fixed by adding some control chars to the translated string.

Reply:
The only improvement for RTL, I see here, is wrapping time in the `TS->format_number(get_time_string(...))` to convert it to localized number system (e.g. eastern Arabic numerals).